### PR TITLE
fix(useElementSize): make directive only registered once

### PR DESCRIPTION
**Pull request by webfansplz**
### Description

Already tested.

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useElementSize/directive.ts [hunk 1]
```diff
@@ -11,13 +11,22 @@ type BindingValueFunction = (size: ElementSize) => void
...
 
-export const vElementSize: FunctionDirective<
+export const vElementSizeHandler = (): FunctionDirective<
 HTMLElement,
 BindingValueFunction | BindingValueArray
-> = (el, binding) => {
-  const handler = typeof binding.value === 'function' ? binding.value : binding.value?.[0]
-  const options = (typeof binding.value === 'function' ? [] : binding.value.slice(1)) as RemoveFirstFromTuple<BindingValueArray>
+> => {
+  let mountedEle: HTMLElement | null = null
```
Comment (antfu) [inline packages/core/useElementSize/directive.ts:18]: What if you apply the directive to multiple elements? I guess we need to maintain a weakset, and remove it when the element unmounted

Comment (webfansplz) [reply packages/core/useElementSize/directive.ts:18]: Good catch , but the functional directive callback only triggered when element mounted and updated , seems hard to remove it.

Comment (webfansplz) [reply packages/core/useElementSize/directive.ts:18]: Maybe we could migrate it to object directives and provide a generic hook to be compatible with Vue 2, what do you think ?

Comment (antfu) [reply packages/core/useElementSize/directive.ts:18]: Yeah I think object style would be better

**Pull request merged**